react/modules/tasks/final.md

```diff
@@ -248,82 +246,73 @@ CORS issues or network timeouts should be shown as errors, whereas 4xx and 5xx H
+Code generation should support at least the following:
 
-```
-+----------------------------------------------------+
-|                      Header                        |
-|  +----------------------------------------------+  |
-|  | [Logo] | Language Toggle | [Sign Out] |      |  |
-|  +----------------------------------------------+  |
-+----------------------------------------------------+
-|                                                    |
-|   +------------------ GraphiQL Client -----------+ |
-|   | Endpoint URL: [URL Input]                    | |
-|   | SDL URL: [URL Input]                         | |
-|   | Headers: [Add Header Button]                 | |
-|   | +-----------------------------------------+  | |
-|   | | Header Key | Header Value               |  | |
-|   | +-----------------------------------------+  | |
-|   | Query: [GraphQL Query Editor]                | |
-|   | Variables: [Variables Editor]                | |
-|   +----------------------------------------------+ |
-|                                                    |
-|   +----------------- Response -------------------+ |
-|   | Status: [HTTP Status Code]                   | |
-|   | Body: [Read-Only JSON Viewer]                | |
-|   +--------------------------------------------+   |
-|   Documentation: [Visible if SDL Response Success] |
-+----------------------------------------------------+
-|                      Footer                        |
-|    [GitHub Link] | Year | [Course Logo]            |
-+----------------------------------------------------+
-```
+- curl
+- JavaScript (Fetch api)
+- JavaScript (XHR)
+- NodeJS
+- Python
+- Java
+- C#
+- Go
+  Please, check the [Postman code generators](https://www.npmjs.com/package/postman-code-generators). Although you can implement the solution on your own, or use any other 3rd party libs.
 
-#### Routing on GraphiQL client
+### Variables
 
-Similar to the RESTfull client, the request infromation should be provided via the url, with the only exception - all the GraphQL queries are POST queries, so to distinguish it from the usual REST queries, you should use GRAPHQL as a first route parameter:
-`http://localhost:5137/GRAPHQL/{endpointUrlBase64encoded}/{bodyBase64encoded}?header1=header1value&header2=header2value...`
+- This route should be private.
+- Variables code should be lazy-loaded (so unauthenticated user won't download the code).
+- Variables should be stored in the browser's local storage.
+- Application should load previously saved variables on user's successful authentication.
+- Variable value can be inserted either in the url, or request body, or header value.
+- To indicate that the variable should be used, user can specify it as {{variableName}}.
+  Variable in the url:
+  `https://wwww.npmjs.com/{{packageName}}`
+  Variable in the header:
+  `{{APPLICATION_JSON}}`
+  Variable in the JSON:
+  ```json
+  {
+    "foo": "{{BAR}}"
+  }
+  ```
```

Comment:
```suggestion
- To indicate that the variable should be used, user can specify it as {{variableName}}.
  
  Variable in the url:
  `https://wwww.npmjs.com/{{packageName}}`
  
  Variable in the header:
  `{{APPLICATION_JSON}}`
  
  Variable in the JSON:
  ```json
  {
    "foo": "{{BAR}}"
  }
  ```
```

Reply:
Added suggested changes in the fix commit.